internal/server/RoundLoop.go

```diff
@@ -77,48 +95,56 @@ func (s *Server) RunBikeSwitch(gameState GameStateDump) {
 func (s *Server) HandleKickoutProcess() []uuid.UUID {
 	allKicked := make([]uuid.UUID, 0)
 	for _, bike := range s.GetMegaBikes() {
-		agentsVotes := make([]uuid.UUID, 0)
+		agents := bike.GetAgents()
+		if len(agents) != 0 {
 
-		// the kickout process only happens democratically in level 0 and level 1
-		switch bike.GetGovernance() {
-		case utils.Democracy:
-			// make map of weights of 1 for all agents on bike
-			agents := bike.GetAgents()
-			weights := make(map[uuid.UUID]float64)
-			for _, agent := range agents {
-				weights[agent.GetID()] = 1.0
-			}
+			agentsVotes := make([]uuid.UUID, 0)
 
-			// get which agents are getting kicked out
-			agentsVotes = bike.KickOutAgent(weights)
+			// the kickout process only happens democratically in level 0 and level 1
+			switch bike.GetGovernance() {
+			case utils.Democracy:
+				// make map of weights of 1 for all agents on bike
+				agents := bike.GetAgents()
+				weights := make(map[uuid.UUID]float64)
+				for _, agent := range agents {
+					weights[agent.GetID()] = 1.0
+				}
 
-		case utils.Leadership:
-			// get the map of weights from the leader
-			leader := s.GetAgentMap()[bike.GetRuler()]
-			weights := leader.DecideWeights(utils.Kickout)
-			// get which agents are getting kicked out
-			agentsVotes = bike.KickOutAgent(weights)
+				// get which agents are getting kicked out
+				agentsVotes = bike.KickOutAgent(weights)
 
-		case utils.Dictatorship:
-			// in level 2 only the ruler can kick out people
-			dictator := s.GetAgentMap()[bike.GetRuler()]
-			agentsVotes = dictator.DecideKickOut()
-		}
+			case utils.Leadership:
+				// get the map of weights from the leader
+				ruler := bike.GetRuler()
+				fmt.Printf("the ruler is %s \n", ruler)
+				leader := s.GetAgentMap()[bike.GetRuler()]
+				weights := leader.DecideWeights(utils.Kickout)
+				// get which agents are getting kicked out
```

Comment:
I suppose leader could be malevolent and never get kicked out if he sets all other agent weights to zero

Reply:
mhh yeah I think that coudl be an interesting behaviour